### DISCUSSION TITLE
Which encryption is in use?

### DISCUSSION THREAD

**Discussion by JoeyVisbeen** [pglombardo/PasswordPusher#402] [Q&A] [answered]
I was looking around the source code and couldn't find where encryption is applied.
So my question is what kind of encryption do you use for your data at rest?

**Comment by pglombardo** [comment, marked as answer]
Hi @JoeyVisbeen - Password Pusher uses [lockbox](https://github.com/ankane/lockbox) (using the AES-GCM algorithm) to encrypt sensitive fields before they are stored.  It acts as an ActiveRecord middle layer to passively encrypt selected fields (e.g. `payload`, `note`).

When hosting your own private instance, we suggest setting your own encryption key.

A bit more documentation on encryption is available here:

* https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#application-encryption
* https://pwpush.com/en/pages/generate_key

I hope this answers your question.  If there is anything else, let me know.

**Comment by pglombardo** [comment]
We now have a page that explains the security of Password Pusher: https://docs.pwpush.com/docs/security/
